### PR TITLE
Preference survey #139

### LINKED ISSUE #139: Preference Survey updates
Required: Support professor type (research or teaching) and number of fall, summer, spring courses

stretch goal: selectable courses to fill out preferences for, not just a long list.

### PR DESCRIPTION
**Pull request by Adamk8**
I did some testing, apparently the backend isn't retrieving preferences properly so the professors page will not reflect the survey yet. The submission is succeeds, we will have to do some more testing later. 

- Added support for choosing Number of Courses per semester 
- Removed the non-teaching term option and replaced with a default value 
- Changed the way that professor chooses the courses they want to choose, now a multi select with an option to select all.
- Add a catch for professor display name being null

Closes #139

### CODE REVIEW

**Review by nicudy — APPROVED** [review]
Looks good!

**Review by AndrewPolanyi — APPROVED** [review]
looks good to me!

File: src/components/Survey/SurveyCourseList.tsx [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    courses: CourseOption[],
+    courseToFind: PreferenceInterface
+  ) => {
+    const filtered = courses.filter((course) => {
+      return (
+        course.value.subject === courseToFind.subject &&
+        course.value.code === courseToFind.code
+      );
+    });
+    return filtered.length > 0;
```
Comment (jayden-chan) [inline src/components/Survey/SurveyCourseList.tsx]: You should replace this with this:
```suggestion
    return courses.some(course => 
      couse.value.subject === courseToFind.subject &&
      course.value.code === courseToFind.code
    );
```

Using `filter` here creates an unnecessary copy of the original array

Comment (Adamk8) [reply src/components/Survey/SurveyCourseList.tsx]: that makes sense, thanks

**Pull request merged**